### PR TITLE
tools/tpm2_nvundefine: Redo tools layout

### LINKED ISSUE #2423: tools/tpm2_nvundefine: Redo tool layout to standard format

### LINKED ISSUE #2519: tools/tpm2_nvundefine: Enable support to specify non-auth sessions for audit and encryption

### LINKED ISSUE #2614: tools/tpm2_nvundefine: Enable rpHash for session-audit support

### PR DESCRIPTION
**Pull request by idesai**
Fixes #2423
Fixes #2614 
Fixes #2519 

Signed-off-by: Imran Desai <imran.desai@intel.com>

### CODE REVIEW

File: doc/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
...
 
+  * tpm2_nvundefine:
+      - Added option **\--rphash**=_FILE_ to specify file path to record the
+        hash of the response parameters. This is commonly termed as rpHash.
+      - Added option **-S**, **\--session** to specify to specify an auxiliary
+        session for auditing and or encryption/decryption of the parameters.
+      - Added option **-n**, **\--name** to specify the name of the nvindex in
+        hex bytes. This is used when cpHash is calculated without dispatching
+        the TPM2_NV_Write command to the TPM.
+
```
Comment (idesai) [inline doc/CHANGELOG.md:13]: must update the changelog with yet another option **--with-policydelete**

File: man/tpm2_nvundefine.1.md [hunk 2]
```diff
@@ -37,13 +37,35 @@ default value for `-C` is the "owner" hierarchy when `TPMA_NV_POLICY_DELETE` is
...
+  * **\--rphash**=_FILE_
+
+    File path to record the hash of the response parameters. This is commonly
+    termed as rpHash.
+
+
+  * **-n**, **\--name**=_FILE_:
+
+    The name of the NV index that must be provided when only calculating the
+    cpHash without actually dispatching the command to the TPM.
```
Comment (idesai) [inline man/tpm2_nvundefine.1.md:69]: must update with option **--with-policydelete**

**Comment by idesai** [comment]
must add a test for cphash with tcti=none
